flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/CompiledPlan.java

```diff
@@ -59,4 +62,12 @@ default void writeToFile(Path path) throws IOException {
     /** Write this plan to a file using the JSON representation. */
     void writeToFile(Path path, boolean ignoreIfExists)
             throws IOException, UnsupportedOperationException;
+
+    // --- Accessors
+
+    /** Get the flink version used to compile the plan. */
+    String getFlinkVersion();
```

Comment:
I don't think so, as these are different. This flink version comes directly from https://github.com/apache/flink/blob/4af7f825a1bcbed0a6fda02c9445d9288a175e30/flink-runtime/src/main/resources-filtered/.flink-runtime.version.properties, so it might contain also SNAPSHOT versions from what i can tell, right?